src/_pytest/_code/code.py

```diff
@@ -343,10 +344,10 @@ def f(cur: TracebackType) -> Iterable[TracebackEntry]:
 
     def cut(
         self,
-        path: Optional[Union[Path, str]] = None,
+        path: Optional[Union["os.PathLike[str]", str]] = None,
```

Comment:
This part also fixes the `sybil` failure - it passes a string absolute path to `cut`, which used to work because `py.path.local('/foo') == '/foo'` but `Path('/foo') != '/foo'`. So the code now compares strings.

Even if we go with the breaking change option we should get this included in pytest 7.0 (this part is non breaking)